src/sentry/sentry_apps/token_exchange/refresher.py

```diff
@@ -30,22 +30,35 @@ class Refresher:
     user: User
 
     def run(self) -> ApiToken:
-        with transaction.atomic(router.db_for_write(ApiToken)):
-            try:
+        try:
+            token = None
+            with transaction.atomic(router.db_for_write(ApiToken)):
                 self._validate()
                 self.token.delete()
 
                 self._record_analytics()
-                return self._create_new_token()
-            except (SentryAppIntegratorError, SentryAppSentryError):
-                logger.info(
-                    "refresher.context",
-                    extra={
-                        "application_id": self.application.id,
-                        "refresh_token": self.refresh_token[-4:],
-                    },
+                token = self._create_new_token()
+                return token
+        except OperationalError as e:
```

Comment:
Instead of catching an `OperationalError` here, I wonder if we should instead catch and raise this in the `outbox_context` decorator as a distinct error.

Reply:
I tried to do this via adding [`OutboxFlushError`](https://github.com/getsentry/sentry/pull/91507/files#diff-b7818151e6f8423eab2449fcf154c350f7a08c44a89d64d04a3942cafaf80a09R198-R206) at the `on_commit` process since that lambda is responsible for all actions after the database has committed. 

I was having trouble understanding what the `test_outbox` tests do & how to write one for the error handling changes. (I also didn't want to spend too much more time digging so I only made a test for refresher.py :bufo-hide:)

Lmk thoughts /